designs/2021-init-command-eslint-cli/README.md

```diff
@@ -0,0 +1,153 @@
+We will make use of tools like `npm` to simply run `npm init @eslint/config` or using `npx` to run `npx  @eslint/create-config` or using `yarn` to run `yarn create @eslint/config` single time for a project instead of having it in the core project.
+
+## Detailed Design
+
+<!--
+   This is the bulk of the RFC.
+
+   Explain the design with enough detail that someone familiar with ESLint
+   can implement it by reading this document. Please get into specifics
+   of your approach, corner cases, and examples of how the change will be
+   used. Be sure to define any new terms in this section.
+-->
+
+The implementation is mainly migrating the code from main repo to `@eslint/create-config`.
+
+The approach would be following these steps
+
+### 1. Remove eslint auto-config
+
+- Remove auto-config(`eslint/lib/init/autoconfig.js`), and its tests(`eslint/tests/lib/init/autoconfig.js`).
+- Remove dependency `progress`, as it's no longer used in production.
+
+### 2. Move `eslint --init` related files to a separate repo
+
+- Move the `eslint/lib/init/*` to the new repo's `lib/*` directory
+- Add the following `dependencies` in `package.json`
+
+  - `enquirer`
+  - `progress`
+  - `semver`
+  - `lodash`
+  - `json-stable-stringify-without-jsonify`
+  - `cross-spawn`
+  - `debug`
+  - `@eslint/eslintrc`
+  - `eslint` (to be removed)
+  - `espree` (to be removed)
+
+- for all those modules coming inside `../**/*`, it would be replaced using `eslint/**/*`
+- For `tests`, move the `tests/lib/init` to new repo's `tests/lib/`
+- Add the following `devDependencies` in `package.json`
+
+  - `chai`
+  - `sinon`
+  - `js-yaml`
+  - `proxyquire`
+  - `shelljs`
+
+It was almost done: [aladdin-add/eslint-create](https://github.com/aladdin-add/eslint-create).
+I will make a PR once the official repo is created later.
+
+### 3. Remove the usage of eslint & espree
+
+eslint was used:
+https://github.com/aladdin-add/eslint-create/blob/c27509cbbaea7c846fd9d7a373feece5254ac8d8/lib/config-file.js#L86
+
+It can be removed by using the local installed eslint, and switch to the new `Linter` api.
+
+espree was used:
+https://github.com/aladdin-add/eslint-create/blob/c27509cbbaea7c846fd9d7a373feece5254ac8d8/lib/config-initializer.js#L161
+
+It can be removed after https://github.com/eslint/espree/issues/495 get landed.
+
+### 4. Release the new package
+
+### 5. Update eslint repo
+
+- Remove `eslint --init` related files.
+- the following dependencies can be removed in eslint repo:
+  - enquirer
+  - semver
```

Comment:
I think we will also need to add `@eslint/create-config` as a dependency?

Reply:
That's not necessary because npm will automatically download and install that when we run `npm init @eslint/config`.